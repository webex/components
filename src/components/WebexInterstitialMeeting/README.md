# Webex Interstitial Meeting Component

Webex Interstitial meeting component displays the [Webex Local Media](../WebexLocalMedia) and [Webex Meeting Info](../WebexMeetingInfo) components for a meeting.

<p align="center">
  <img src="./WebexInterstitialMeeting.png" alt="Default Webex Interstitial Meeting" />
</p>

## Preview

To see all the different possible states of the Webex Interstitial Meeting component, you can run our Storybook:

```shell
  npm start
```

## Embed

1.  Create a component adapter from which the data will be retrieved (See [adapters](../../adapters)). For instance:

    ```js
    const jsonAdapter = new WebexJSONAdapter(jsonData);
    ```

2.  Create a component instance by passing the person ID as a string and
    enclose it within [a data provider](../WebexDataProvider/WebexDataProvider.js)
    that takes the [component data adapter](../../adapters/WebexJSONAdapter.js) that we created previously

<<<<<<< HEAD
    ```js
    <WebexDataProvider adapter={jsonAdapter}>
      <WebexInterstitialMeeting meetingDestination="meetingDestination" onMeeting={this.onMeeting} />
    </WebexDataProvider>
    ```
=======
        ```js
        <WebexDataProvider adapter={jsonAdapter}>
          <WebexInterstitialMeeting meetingDestination="meetingDestination"  setMeeting="this.setMeeting"/>
        </WebexDataProvider>
        ```
>>>>>>> 21b376e... docs(InterstitialMeeting): add README

The component knows how to manage its data. If anything changes in the data source that the adapter manages, the component will also update on its own.
