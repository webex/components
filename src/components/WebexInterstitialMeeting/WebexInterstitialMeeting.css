.interstitial-meeting {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.interstitial-meeting .meeting-info {
  position: absolute;
  align-self: flex-start;
  width: 100%;
  height: 25%;

  background-color: rgba(255, 255, 255, 0.8);
}
