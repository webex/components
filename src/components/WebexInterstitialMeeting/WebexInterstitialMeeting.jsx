import React from 'react';
import PropTypes from 'prop-types';
import {Spinner} from '@momentum-ui/react';

import {WEBEX_COMPONENTS_CLASS_PREFIX} from '../../constants';

import WebexLocalMedia from '../WebexLocalMedia/WebexLocalMedia';
import WebexMeetingInfo from '../WebexMeetingInfo/WebexMeetingInfo';

/**
 * Webex Interstitial component displays the user's local video and
 * meeting info, in preparation to joining a meeting.
 *
 * @param {object} props
 * @returns {object} JSX of the component
 */
export default function WebexInterstitialMeeting({meetingID}) {
  return (
    <div className={`${WEBEX_COMPONENTS_CLASS_PREFIX}-interstitial-meeting`}>
      {meetingID ? (
        <>
          <WebexMeetingInfo className="interstitial-meeting-info" meetingID={meetingID} />
          <WebexLocalMedia meetingID={meetingID} />
        </>
      ) : (
        <Spinner />
      )}
    </div>
  );
}

WebexInterstitialMeeting.propTypes = {
  meetingID: PropTypes.string.isRequired,
};
