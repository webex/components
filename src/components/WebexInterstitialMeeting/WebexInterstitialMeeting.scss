.#{$WEBEX_COMPONENTS_CLASS_PREFIX}-interstitial-meeting {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  min-width: $WEBEX_MEETING_MIN_WIDTH;

  .media-container {
    flex: 1;
    align-self: stretch;
    display: flex;
    flex-direction: column;
  }

  .interstitial-media-container {
    flex: 1;
    min-height: 0;
    align-self: stretch;
    display: flex;
    flex-direction: column;
    position: relative;
  }

  .interstitial-media {
    flex: 1;
    min-height: 0;
    align-self: stretch;
  }

  .interstitial-meeting-info {
    margin-bottom: 1rem;
  }
}
