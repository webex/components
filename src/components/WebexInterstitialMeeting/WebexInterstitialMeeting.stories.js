import React from 'react';
import {storiesOf} from '@storybook/react';

import jsonData from '../../data';
import {WebexJSONAdapter} from '../../adapters';
import {WebexInterstitialMeeting, WebexDataProvider} from '../';

// Setup for the stories
const stories = storiesOf('Webex Interstitial Meeting', module);
const webexAdapter = new WebexJSONAdapter(jsonData);
const wrapperStyle = {height: '500px', width: '800px', border: '1px solid black'};

stories.add('loading', () => (
  <div style={wrapperStyle}>
    <WebexDataProvider adapter={webexAdapter}>
      <WebexInterstitialMeeting meetingID="" />
    </WebexDataProvider>
  </div>
));

stories.add('video enabled', () => {
  jsonData.meetings.localMedia.localVideo = {};

  return (
    <div style={wrapperStyle}>
      <WebexDataProvider adapter={webexAdapter}>
        <WebexInterstitialMeeting meetingID="localMedia" />
      </WebexDataProvider>
    </div>
  );
});

stories.add('video disabled', () => {
  jsonData.meetings.localMedia.localVideo = null;

  return (
    <div style={wrapperStyle}>
      <WebexDataProvider adapter={webexAdapter}>
        <WebexInterstitialMeeting meetingID="localMedia" />
      </WebexDataProvider>
    </div>
  );
});
