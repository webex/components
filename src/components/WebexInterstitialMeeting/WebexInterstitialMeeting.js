import React from 'react';
import PropTypes from 'prop-types';
import {Spinner} from '@momentum-ui/react';

import {WEBEX_COMPONENTS_CLASS_PREFIX} from '../../constants';

import {WebexLocalMedia, WebexMeetingInfo} from '..';

import './WebexInterstitialMeeting.scss';

/**
 * Webex Interstitial component displays the user's local video and
 * meeting info, in preparation to joining a meeting.
 *
 * @param {object} props
 * @returns {object} JSX of the component
 */
export default function WebexInterstitialMeeting({meetingID}) {
  return (
    <div className={`${WEBEX_COMPONENTS_CLASS_PREFIX}-interstitial-meeting`}>
      {meetingID ? (
        <React.Fragment>
          <WebexMeetingInfo meetingID={meetingID} />
          <WebexLocalMedia meetingID={meetingID} />
        </React.Fragment>
      ) : (
        <Spinner />
      )}
    </div>
  );
}

WebexInterstitialMeeting.propTypes = {
  meetingID: PropTypes.string.isRequired,
};
