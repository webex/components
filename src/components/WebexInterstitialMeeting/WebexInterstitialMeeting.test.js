import React from 'react';

import WebexInterstitialMeeting from './WebexInterstitialMeeting';

jest.mock('../hooks/useMeeting');

describe('Webex Interstitial Meeting component', () => {
  describe('snapshot', () => {
    test('matches snapshot of loading while meeting is created', () => {
      expect(shallow(<WebexInterstitialMeeting meetingDestination="" />)).toMatchSnapshot();
    });

    test('matches snapshot of meeting destination', () => {
      expect(shallow(<WebexInterstitialMeeting meetingDestination="localMedia" />)).toMatchSnapshot();
    });
  });
});
