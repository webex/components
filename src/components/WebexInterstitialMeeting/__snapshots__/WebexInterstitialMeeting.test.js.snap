// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webex Interstitial Meeting component snapshot matches snapshot of loading while meeting is created 1`] = `
<div
  className="wxc-interstitial-meeting"
>
  <Spinner
    className=""
    color="black"
    percentage={null}
    showCheck={false}
    showPercentage={false}
    size={36}
  />
</div>
`;

exports[`Webex Interstitial Meeting component snapshot matches snapshot of meeting destination 1`] = `
<div
  className="wxc-interstitial-meeting"
>
  <WebexMeetingInfo
    meetingID="localMedia"
  />
  <WebexLocalMedia
    meetingID="localMedia"
  />
</div>
`;
